operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/rate/RateLimiter.java

```diff
@@ -3,20 +3,17 @@
 import java.time.Duration;
 import java.util.Optional;
 
-import io.javaoperatorsdk.operator.processing.event.ResourceID;
+import io.javaoperatorsdk.operator.processing.event.rate.RateLimiter.RateLimitState;
 
-public interface RateLimiter {
+public interface RateLimiter<S extends RateLimitState> {
+  interface RateLimitState {
```

Comment:
Can't this be just an Object? Do we need an explicit interface for this. 
I mean just use `RateLimiter<S>`

Reply:
Otherwise LGTM

Reply:
Yeah, not sure either… Was thinking that perhaps we needed some some of marker interface to be able to trace where this state is used more easily than tracking `Object`… Will review.

Reply:
I think we should keep it this way so that it's more obvious what the associated state is for, even though technically `Object` would work just as well.

Reply:
Can't remember where but we had the same situation, where we decided to go with the object and don't use marker interfaces. 
So I lean more to remove it, but I'm fine having that if you think it has benefits.

Reply:
It also has the advantage of having an interface already in case we need to somehow restrict what can be used…

Reply:
See also https://github.com/java-operator-sdk/java-operator-sdk/pull/1319/files#r917897536